docs/src/manual/models.md

```diff
@@ -39,14 +38,74 @@ and additionally provide methods to solve the model.
 
 ## Attributes
 
-Attributes can be set in different ways:
+Attributes are properties of the model that can be queried and modified. These
+include constants such as the number of variables in a model [`NumberOfVariables`](@ref)),
+and properties of variables and constraints such as the name of a variable
+([`VariableName`](@ref)).
+
+There are four types of attributes:
+
+ * Model attributes (subtypes of [`AbstractModelAttribute`](@ref)) refer to
+   properties of a model.
+ * Optimizer attributes (subtypes of [`AbstractOptimizerAttribute`](@ref)) refer
+   to properties of an optimizer.
+ * Constraint attributes (subtypes of [`AbstractConstraintAttribute`](@ref))
+   refer to properties of an individual constraint.
+ * Variable attributes (subtypes of [`AbstractVariableAttribute`](@ref)) refer
+   to properties of an individual variable.
+
+Some attributes are constants that can be queried by the user but not modified,
```

Comment:
What is `ListOfModelAttributesSet` ? It is not constant but it cannot be modified directly